### ISSUE TITLE
Add support for long polling

### ISSUE THREAD

**Issue by jwulf** [creditsenseau/zeebe-client-node-js#64]
0.21.0 looks set to support long polling in the client. See: https://github.com/zeebe-io/zeebe/pull/2930

**Comment by jwulf** [comment]
# Zeebe e-commerce demo with inventory worker using long polling:

Running the [Zeebe e-commerce demo](https://github.com/jwulf/zeebe-ecommerce-demo) - 1000 workflows, starting 25/second.

## With Simple Monitor enabled

Long poll:
```
Execution time (hr): 62s 662.513835ms
Execution time (hr): 54s 322.733574ms
Execution time (hr): 56s 696.164087ms
```
With 100ms continuous poll:

```
Execution time (hr): 53s 426.178514ms
Execution time (hr): 58s 519.939317ms
Execution time (hr): 90s 257.462871ms
```

The last one is probably due to the high volume of Activate Requests being processed by the exporter. There was another run before that one, but it had failed workflow creations, because the server load was too high. That one was 84s.

## With Broker Only

Long poll:

```
Execution time (hr): 42s 168.803361ms
Execution time (hr): 42s 173.879616ms
Execution time (hr): 42s 288.298501ms
```

With 100ms continuous poll:

```
Execution time (hr): 42s 328.696198ms
Execution time (hr): 42s 249.656159ms
Execution time (hr): 42s 252.437925ms
```

Conclusion: the client performance is the same, but long polling significantly reduces the load on the server.

**Comment by jwulf** [comment]
### Long polling

With Zeebe 0.21 onward, long polling is supported for clients. Rather than polling continuously for work and getting nothing back, a client can poll once and leave the request open until work appears. This reduces network traffic and CPU utilization in the server. Every JobActivation Request is appended to the event log, so continuous polling can significantly impact broker performance, especially when an exporter is loaded (see [here](https://github.com/creditsenseau/zeebe-client-node-js/issues/64#issuecomment-520233275)). The Node client is hardcoded to long poll for ten minute intervals when long polling is enabled.

Long polling for workers is enabled in the ZBClient, like this:

```typescript
const zbc = new ZBClient('serverAddress', {
	longPoll: true,
})

const longPollingWorker = zbc.createWorker(uuid.v4(), 'task-type', handler)
```

**Comment by jwulf** [comment]
There is an issue that may require a deeper refactor. 

If the broker goes away, a long polling client does not reconnect when the broker comes back.

The fast-polling client reconnects, but for some reason the long-polling one stays in a failed state....

**Comment by jwulf** [comment]
Looks like the fast polling one doesn't reliably reconnect either. See here: https://github.com/grpc/grpc/issues/11411#issuecomment-308899967

The Node gRPC does support reconnection, but it doesn't seem to be working.

**Comment by jwulf** [comment]
Depends on https://github.com/zeebe-io/zeebe/issues/2987

**Comment by jwulf** [comment]
Closed by #68
